tests/api/test_flows.py

```diff
@@ -514,6 +516,36 @@ async def test_create_flow_persists_serialized_flow(self, project_id, flow):
         # confirm the keys in the serialized flow match the form we'd expect
         assert persisted_flow.serialized_flow == flow.serialize()
 
+    async def test_create_flow_cleans_up_if_task_creation_fails(
+        self, project_id, flow, monkeypatch
+    ):
+        patched_insert_raises_error = MagicMock(side_effect=Exception())
+        monkeypatch.setattr(
+            "prefect.models.Task.insert_many", patched_insert_raises_error
+        )
+        patched_delete = MagicMock()
+        monkeypatch.setattr("prefect_server.api.flows.delete_flow", patched_delete)
```

Comment:
I think we should assert that it's deleted rather than using a mock.

Reply:
Agreed -- one way to do this would be to lookup the name of this flow in the DB and confirm that it is not present before and after your `create_flow` call.

Reply:
went with Chris' suggestion in the latest commit